src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs

```diff
@@ -95,30 +91,39 @@ private static Location GetLocation(Symbol symbol)
         /// <remarks>
         /// Don't call this one directly - call one of the helpers.
         /// </remarks>
-        protected bool ValidateNameConflictsInScope(Symbol symbol, Location location, string name, DiagnosticBag diagnostics)
+        private bool ValidateNameConflictsInScope(Symbol symbol, Location location, string name, DiagnosticBag diagnostics)
         {
             if (string.IsNullOrEmpty(name))
             {
                 return false;
             }
 
-            bool error = false;
             for (Binder binder = this; binder != null; binder = binder.Next)
             {
                 // no local scopes enclose members
-                if (binder is InContainerBinder || error)
+                if (binder is InContainerBinder)
                 {
-                    break;
+                    return false;
                 }
 
                 var scope = binder as LocalScopeBinder;
-                if (scope != null)
+                if (scope?.EnsureSingleDefinition(symbol, name, location, diagnostics) == true)
                 {
-                    error |= scope.EnsureSingleDefinition(symbol, name, location, diagnostics);
+                    return true;
+                }
+
+                // If shadowing is enabled, avoid checking for conflicts outside of local functions.
+                if (Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticLocalFunctions))
```

Comment:
>Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticLocalFunctions) [](start = 20, length = 71)

It looks like this can be calculated before we enter the loop